### ISSUE TITLE
Is partial fraction expansion supported?

### ISSUE THREAD

**Issue by standarddeviant** [JuliaMath/Polynomials.jl#128]
Hello,
I'd like to perform partial fraction expansion on the division of two polynomials, but I don't see that functionality in Polynomials.jl

There's a BSD version of PFE or "residue" here:
https://github.com/scipy/scipy/blob/2aae8e8a6f9580bf98c5a4b1dc2ce5202c7a427b/scipy/signal/signaltools.py#L1837 

Does anyone know if this is supported in the greater Julia ecosystem?

Many thanks,
Dave

**Comment by jverzani** [comment]
Maybe in SymPy, but not here. A PR would be welcome, but not based on BSD code, unless compatible with the license used here.

**Comment by standarddeviant** [comment]
Okay, that makes sense about the licensing.  I may translate this source (MIT licensed):
https://github.com/stringman45/part_frac/blob/master/part_frac.sage 

If I do, I'll submit a PR.

**Comment by jverzani** [comment]
In case you just want to use something, in `SymPy` the `apart` function does this. If you want to program this, you should look at some of the code in `PolynomialFactors`. It seems that any such functionality might be better suited for that package, than `Polynomials` itself.

**Comment by standarddeviant** [comment]
@jverzani Unfortunately I just saw your response after writing an implementation:
https://github.com/standarddeviant/JOS3.jl/blob/master/src/filters/residue.jl 

I have two questions:
1) Does PolynomialFactors has a good way of handling repeated roots?  I think that's the most brittle part of my implementation.
2) Should I close this issue and open a new one in [`PolynomialFactors`](https://github.com/jverzani/PolynomialFactors.jl) ?

**Comment by standarddeviant** [comment]
@jverzani I made a version of `residue` that uses PolynomialFactors, but it only works for polynomials with Integer coefficients.  Does PolynomialFactors support Float coefficients for polynomials?  I'd be happy to submit a PR to PolynomialFactors for what I have.

**Comment by jverzani** [comment]
For 1, over the integer or rationals, `PolynomialFactors` can identify repeated irreducible factors. However over the reals of complex, the function `agcd` in the `PolynomialZeros` package can be used to find an *approximate* gcd of the polynomial and its derivative to reduce the poly to one with no repeated factors.

**Comment by jverzani** [comment]
No, PolynomialFactors is limited to exact coefficients (integer and rational), the floating point routines are in PolynomialZeros. It would make more sense to be there if you didn't want a standalone package.
